File: src/main/java/no/nav/familie/ks/sak/app/rest/Behandling/grunnlag/søknad/RestAktørArbeidYtelseUtland.java
package no.nav.familie.ks.sak.app.rest.Behandling.grunnlag.søknad;

import no.nav.familie.ks.sak.app.behandling.domene.kodeverk.Standpunkt;
import no.nav.familie.ks.sak.app.behandling.domene.typer.AktørId;

public class RestAktørArbeidYtelseUtland {
    public AktørId aktørId;
    public Standpunkt arbeidIUtlandet;
    public String arbeidIUtlandetForklaring;
    public Standpunkt ytelseIUtlandet;
    public String ytelseIUtlandetForklaring;
    public Standpunkt kontantstøtteIUtlandet;
    public String kontantstøtteIUtlandetForklaring;

    public RestAktørArbeidYtelseUtland(AktørId aktørId, Standpunkt arbeidIUtlandet, String arbeidIUtlandetForklaring, Standpunkt ytelseIUtlandet, String ytelseIUtlandetForklaring, Standpunkt kontantstøtteIUtlandet, String kontantstøtteIUtlandetForklaring) {
        this.aktørId = aktørId;
        this.arbeidIUtlandet = arbeidIUtlandet;
        this.arbeidIUtlandetForklaring = arbeidIUtlandetForklaring;
        this.ytelseIUtlandet = ytelseIUtlandet;
        this.ytelseIUtlandetForklaring = ytelseIUtlandetForklaring;
        this.kontantstøtteIUtlandet = kontantstøtteIUtlandet;
        this.kontantstøtteIUtlandetForklaring = kontantstøtteIUtlandetForklaring;
    }
}

File: src/main/java/no/nav/familie/ks/sak/app/rest/behandling/grunnlag/søknad/RestAktørArbeidYtelseUtland.kt
package no.nav.familie.ks.sak.app.rest.behandling.grunnlag.søknad

import no.nav.familie.kontrakter.ks.søknad.Standpunkt
import no.nav.familie.ks.sak.app.behandling.domene.grunnlag.søknad.AktørArbeidYtelseUtland

data class RestAktørArbeidYtelseUtland(
    val fødselsnummer: String,
    val arbeidIUtlandet: Standpunkt,
    val arbeidIUtlandetForklaring: String,
    val ytelseIUtlandet: Standpunkt,
    val ytelseIUtlandetForklaring: String,
    val kontantstøtteIUtlandet: Standpunkt = Standpunkt.UBESVART,
    val kontantstøtteIUtlandetForklaring: String? = null)

fun AktørArbeidYtelseUtland.toRestAktørArbeidYtelseUtland() = RestAktørArbeidYtelseUtland(
        fødselsnummer = this.fødselsnummer,
        arbeidIUtlandet = this.arbeidIUtlandet,
        arbeidIUtlandetForklaring = this.arbeidIUtlandetForklaring,
        ytelseIUtlandet = this.ytelseIUtlandet,
        ytelseIUtlandetForklaring = this.ytelseIUtlandetForklaring,
        kontantstøtteIUtlandet = this.kontantstøtteIUtlandet,
        kontantstøtteIUtlandetForklaring = this.kontantstøtteIUtlandetForklaring
)
